.github/workflows/ci.yml

```diff
@@ -71,7 +71,10 @@ jobs:
       - uses: software-mansion/setup-scarb@v1.0.1
         with:
           scarb-version: ${{ env.SCARB_VERSION }}
-      - run: cargo test -p cheatnet
+      - name: Run Cheatnet tests
+        run: |
+          echo CHEATNET_RPC_URL=${{ secrets.CHEATNET_RPC_URL }} >> .env
+          cargo test -p cheatnet
```

Comment:
Also will this work for external PRs at all?

Reply:
> Also will this work for external PRs at all?
Depends on what you count as an external PR. If someone forks and runs CI on the fork, the secret is not available.
When someone gets an approval to run CI on our repo - this is where things start to get tricky. We must make sure the output is censored (I'm pretty positive it is) and that no shenanigans in the form of redirecting the secret to file and downloading it works. Pretty solid PRs before allowing the runs should suffice, but we must make sure that those are needed after every new commit (otherwise someone that got such approval could try to do the file redirect thing).

Reply:
I think the current flow is that only approval for the first run is needed (actually its for the first PR, but it can be changed) and then the subsequent runs just work. So this is not ideal.

Though [the docs say](https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks) "Although workflows from forks do not have access to sensitive data such as secrets, they can be an annoyance for maintainers if they are modified for abusive purposes." so it seems this will not work with PRs from forks at all. But it should be verified probably.

Reply:
Conclusion:
- every new commit requires approval
- amended commits too
- force-pushed branch also

We can safely keep our node address in the secrets